ddl/ddl_api.go

```diff
@@ -6616,6 +6617,16 @@ func (d *ddl) AlterTableCache(ctx sessionctx.Context, ti ast.Ident) (err error)
 	if t.Meta().Partition != nil {
 		return errors.Trace(ErrOptOnCacheTable.GenWithStackByArgs("partition mode"))
 	}
+
+	// Initialize the cached table meta lock info in `mysql.table_cache_meta`.
+	// The operation shouldn't fail in most cases, and if it does, return the error directly.
+	// This DML and the following DDL is not atomic, that's not a problem.
+	_, err = ctx.(sqlexec.SQLExecutor).ExecuteInternal(context.Background(),
+		"insert ignore into mysql.table_cache_meta values (%?, 'NONE', 0, 0)", t.Meta().ID)
+	if err != nil {
+		return errors.Trace(err)
+	}
```

Comment:
Is it necessary to remove the non-existent or uncached table?

Reply:
This what I disscussed with @lcwangchao , the additional DML and the DDL operation is not atomic,
so if there are multiple `alter table t cache`  `alter table t nocache` and they interleave with each other,
we can't make sure finally the row is correctly inserted (maybe inserted then deleted by the other DDL unexpectedly)

If we just insert the row but don't remove it, the correctness is fine.
And I'll leave it as an issue and solve it later.
Some possible solutions maybe:
1. Don't insert this row and let StateRemote handle the uninitialized meta lock as 'None'
2. Or move this initialize operation inside the DDL schema change, handle the DDL abnormal carefully
3. Or keep the current strategy and remove the row only when the table is dropped

@tangenta

Reply:
I prefer to combine 1 and 3. On one hand, `StateRemote` treats the uninitialized lock as None. On the other hand, remove the row when the table is dropped to prevent `table_cache_meta` from being too large.